homeassistant/components/weather/__init__.py

```diff
@@ -110,6 +118,14 @@ def state_attributes(self):
         if attribution is not None:
             data[ATTR_WEATHER_ATTRIBUTION] = attribution
 
+        forecast = self.forecast
+        if forecast is not None:
+            for forecast_entry in forecast:
+                forecast_entry[ATTR_FORECAST_TEMP] = self._temp_for_display(
```

Comment:
I'm not sure if I'm understanding this correctly, but isn't it copied already by the call in line 121 (it's the whole forecast instead of each individual entry)?
If it's not, then how can I replace the forecast_entry in the forecast array itself, it seems I can't get it to be replaced or properly returned